core/modules/server/server.js

```diff
@@ -171,7 +171,15 @@ Server.prototype.get = function(name) {
 };
 
 Server.prototype.addRoute = function(route) {
+	// Find out if the route exists
+	let index = this.routes.findIndex((thisRoute) => thisRoute.path.toString() === route.path.toString());
```

Comment:
we don't use `let` and arrow functions. TW is still ES5 atm

Reply:
This isn't correct, because different HTTP METHOD requests can have the same path. I'm pretty sure this isn't backward compatible. 

Furthermore, I think this should be left up to the calling code since the routes list is an instance variable and can certainly be modified by calling code (i.e. `listen` and `ws-server` commands) and child classes. So whoever is calling addRoute can do that check for themself. If this is for the default route loader, I don't think this is how we should fix that conundrum. While this may be a useful feature, I don't think this way of doing it really follows TiddlyWiki standard practice. I've listed below several ways that plugins can legitimately or illegitimately override already loaded routes.

If you are using the `--listen` command, then I would recommend making use of the `th-server-command-post-start` hook, which has the server instance as its first argument. If you wrote your own command you can access the server instance there. 

Here are the different options I can think of which plugins can currently use to remove routes which they want to replace. 

- overloading the addRoute function (`var original = server.addRoute; server.addRoute = function() { original(...arguments); }`), or manually removing the offending routes that you want to override (`server.routes.splice(index, 1)`). 
- overriding the shadow tiddler for those routes by including a tiddler with the same name in your plugin that either creates a new route or doesn't include the `module-type` field. This will override the shadow tiddler from any previous plugin according to standard plugin loading order. 

The second option is preferred for feature plugins that just add one feature. If I'm not mistaken, in this particular case this will be used for the core Bob plugin in the root datafolder so interfering with other plugins is not a concern.

Reply:
Reverted.